src/BL/CurrencyManager.java

```diff
@@ -18,28 +22,56 @@ public CurrencyManager() {
      * Метод getAllCurrencies - возвращает все валюты и курсы их обмена по отношению к дефолтной валюте.
      * Использует метод read() класса CurrencyExchangeHandler.
      * Метод возвращает HashMap с валютами и их курсами.
-    **/
-     public HashMap<CurrencyTitle, Double> getAllCurrency () {
+     **/
+    public HashMap<CurrencyTitle, Double> getAllCurrency() {
         return data.read();
     }
 
-    /** Метод calculateRate - рассчитывает курс обмена между двумя валютами
-     * Параметры метода: initialCurrency начальная валюта, resultCurrency  конечная валюта
+    /**
+     * Метод calculateRate - рассчитывает курс обмена между двумя валютами
+     * Параметры метода: initialCurrency начальная валюта, resultCurrency конечная валюта
      * Метод возвращает переменную типа double - курс обмена между заданными валютами
      **/
-    public double calculateRate (CurrencyTitle initialCurrency, CurrencyTitle resultCurrency){
+    public double calculateRate(CurrencyTitle initialCurrency, CurrencyTitle resultCurrency) {
         HashMap<CurrencyTitle, Double> exchangeRates = data.read();
         return exchangeRates.get(resultCurrency) / exchangeRates.get(initialCurrency);
-        //TODO - предусмотреть случай, когда курс = 0
     }
 
-    /** Метод calculateTotalAmount - рассчитывает конечную сумму после обмена.
+    /**
+     * Метод calculateTotalAmount - рассчитывает конечную сумму после обмена.
      * Параметры метода: InitialCurrencyAmount - сумма обмена, initialCurrency начальная валюта, resultCurrency конечная валюта
      * Метод возвращает переменную типа double - конечная сумма после обмена
-    **/
-    public double calculateTotalAmount (double InitialCurrencyAmount,CurrencyTitle initialCurrency, CurrencyTitle resultCurrency) {
+     **/
+    public double calculateTotalAmount(double InitialCurrencyAmount, CurrencyTitle initialCurrency, CurrencyTitle resultCurrency) {
         return calculateRate(initialCurrency, resultCurrency) * InitialCurrencyAmount;
+    }
 
+    /**
+     * Метод getCurrencyTitles - сравнивает значения из enum CurrencyTitle
+     * и значения из файла с курсами валют
+     * Метод возвращает массив из CurrencyTitle, в котором содержаться только общие
+     * элементы
+     **/
+    public List<CurrencyTitle> getCurrencyTitles() {
+        CurrencyTitle[] currencyTitlesEnum = CurrencyTitle.values();
+        HashMap<CurrencyTitle, Double> currency_file = data.read();
+        CurrencyTitle[] currencyTitlesFile = currency_file.keySet().toArray(new CurrencyTitle[0]);
+
```

Comment:
The method can be simplified) I remembered that we cannot put any invalid values into the HashMap keys if there is no such value in the Enum. Accordingly, here we can return the keys from the HashMap and this will be enough.
Sorry, my fault